src/sdg/producten/management/commands/autofill.py

```diff
@@ -0,0 +1,75 @@
+from django.conf import settings
+from django.core.management import BaseCommand
+from django.db.models import Count
+
+from sdg.core.models import ProductenCatalogus, UniformeProductnaam
+from sdg.producten.models import (
+    GeneriekProduct,
+    LocalizedGeneriekProduct,
+    LocalizedProduct,
+    Product,
+    ProductVersie,
+)
+
+
+class Command(BaseCommand):
+    help = "Generate generic and specific (reference) products and fill catalogs where needed."
+
+    def handle(self, **options):
+        # Get all catalogs that can be filled and store matching fields.
+        catalogs = []
+        for catalog in ProductenCatalogus.objects.filter(
+            autofill=True, is_referentie_catalogus=True
+        ):
+            fields = [f.lower() for f in catalog.autofill_upn_filter]
+            if fields:
+                catalogs.append((catalog, fields))
+
+        # Bail early if no catalogs.
+        if not catalogs:
+            self.stdout.write("No catalogs found to autofill.")
+
+        # Iterate over the UPL, create generic (+localized) and specific
+        # (+localized, +version) products if they don't exist yet.
+
+        # NOTE: New languages are not created automatically. There's no need
+        # for this feature at the moment.
+        for upn in UniformeProductnaam.objects.annotate(Count("generieke_producten")):
+            if upn.generieke_producten__count == 0:
+                generic_product = GeneriekProduct.objects.create(upn=upn)
+                LocalizedGeneriekProduct.objects.localize(
+                    instance=generic_product, languages=settings.SDG_ACTIVE_LANGUAGES
+                )
+                self.stdout.write(f'Created new generic product for "{upn}".')
+
+                generic_products = [generic_product]
+            else:
+                generic_products = GeneriekProduct.objects.filter(upn=upn)
```

Comment:
```suggestion
            if upn.generieke_producten__count == 0:
                generic_product = GeneriekProduct.objects.create(upn=upn)
                LocalizedGeneriekProduct.objects.localize(
                    instance=generic_product, languages=TaalChoices.get_available_languages(),
                )
                self.stdout.write(f'Created new generic product for "{upn}".')
                generic_products = [generic_product]
            else:
                generic_products = upn.generieke_producten
```

As we already have TaalChoices to set the existing available languages it should be better to use it instead.